api/kubernetes/deploy_via_helm.sh

```diff
@@ -205,19 +218,148 @@ EOF
+  # work if the new pod is scheduled on a different node.
+  # It is possible to force the replicas scheduling on a same node, but this requires that node
+  # to have a minimum of 4Gi additional memory available, which is the default memory request
+  # set by Argo.
+  type: Recreate
+networkPolicy:
+  # Enabling networking policy returns the following error: unable to recognize "": no matches for kind "NetworkPolicy" in version "networking.k8s.io/v1beta1"
+  # => will use the 'networking/traffic-allowed: "yes"' label instead to allow traffic.
+  enabled: false
+  allowExternal: true
+podLabels:
+  networking/traffic-allowed: "yes"
+
+EOF
+
+helm repo add minio https://helm.min.io/
+helm upgrade --install ${MINIO_RELEASE_NAME} minio/minio --namespace ${NAMESPACE} --version ${MINIO_VERSION} --values values-minio.yaml
+
+# Postgres
+cat <<EOF > values-postgresql.yaml
+auth:
+  username: "${ARGO_POSTGRESQL_USER}"
+  password: "${ARGO_POSTGRESQL_PASSWORD}"
+  database: ${ARGO_DATABASE}
+primary:
+  podLabels:
+    "networking/traffic-allowed": "yes"
+
+EOF
+
+helm repo add bitnami https://charts.bitnami.com/bitnami
+helm upgrade --install -n ${NAMESPACE} ${POSTGRES_RELEASE_NAME} bitnami/postgresql --version ${POSTGRESQL_VERSION} --values values-postgresql.yaml
+
+export ARGO_POSTGRESQL_SECRET=argo-postgres-config
+cat <<EOF > postgres-secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app: postgres
+  name: ${ARGO_POSTGRESQL_SECRET}
+stringData:
+  password: ${ARGO_POSTGRESQL_PASSWORD}
+  username: ${ARGO_POSTGRESQL_USER}
+type: Opaque
+
+EOF
+kubectl apply -n ${NAMESPACE} -f postgres-secret.yaml
+
 # Argo
-export ARGO_RELEASE_NAME=argo
-export ARGO_RELEASE_NAMESPACE="${NAMESPACE}"
-export ARGO_POSTGRESQL_PASSWORD="$3"
-helm pull oci://ghcr.io/cosmo-tech/csm-argo-chart --version "${CHART_PACKAGE_VERSION}" --untar
-# Default memory request in MinIO Chart is 4Gi, which may not work in clusters with lower resources
-envsubst < ./csm-argo/values.yaml | \
-    helm upgrade --install "${ARGO_RELEASE_NAME}" ./csm-argo \
-        --namespace "${NAMESPACE}" \
-        --values - \
-        --set argo.minio.resources.requests.memory="${ARGO_MINIO_REQUESTS_MEMORY:-4Gi}" \
-        --set argo.minio.accessKey="${ARGO_MINIO_ACCESS_KEY:-}" \
-        --set argo.minio.secretKey="${ARGO_MINIO_SECRET_KEY:-}"
+cat <<EOF > values-argo.yaml
+workflow:
+  serviceAccount:
+    create: true
+    name: workflow
+  rbac:
+    create: true
+executor:
+  env:
+  - name: RESOURCE_STATE_CHECK_INTERVAL
+    value: 1s
+artifactRepository:
+  archiveLogs: true
+  s3:
+    bucket: ${ARGO_BUCKET_NAME}
+    endpoint: ${MINIO_RELEASE_NAME}.${NAMESPACE}.svc.cluster.local:9000
+    insecure: true
+    accessKeySecret:
+      name: ${MINIO_RELEASE_NAME}
+      key: accesskey
+    secretKeySecret:
+      name: ${MINIO_RELEASE_NAME}
+      key: secretkey
+server:
+  extraArgs:
+  - --auth-mode=server
+  secure: false
```

Comment:
I don't know what the `secure: false` and I'm too afraid to ask :p
Could you please check that this configuration is PROD compliant?

Reply:
I think to that to secure the workflows we can configure tls and miinio
http://pipekit.io/blog/production-install-of-argo-workflows
https://blog.argoproj.io/practical-argo-workflows-hardening-dd8429acc1ce

Reply:
It is needed to avoid credential between the api and argo